website/addons/wiki/views.py

```diff
@@ -356,6 +363,44 @@ def project_wiki_edit_post(auth, wname, **kwargs):
         ret = {'status': 'success'}
     return ret, http.FOUND, None, redirect_url
 
+@must_be_valid_project  # injects node or project
+@must_have_permission('admin')
+@must_not_be_registration
+@must_have_addon('wiki', 'node')
+def edit_wiki_permissions(**kwargs):
+    node = kwargs['node'] or kwargs['project']
+    wiki_settings = node.get_addon('wiki')
+    permissions = kwargs.get('permissions')
+    auth = kwargs['auth']
+
+    if permissions is None:
+        raise HTTPError(http.BAD_REQUEST)
+
+    if wiki_settings is None:
+        raise HTTPError(http.BAD_REQUEST)
+
+    try:
+        edited = wiki_settings.set_editing(permissions, auth, node, True, True)
+    except AttributeError as e:
+        raise HTTPError(http.BAD_REQUEST, data=dict(
+            message_short=e.message
+        ))
+    else:
```

Comment:
Another thing - you're using the return value of `set_editing` here. Why not just raise an appropriate exception inside that method, then bubble up the message if it exists?